packages/koenig-lexical/src/plugins/TKPlugin.jsx

```diff
@@ -192,52 +248,48 @@ export default function TKPlugin({onCountChange = () => {}}) {
         };
     }, []);
 
-    useLexicalTextEntity(
+    // TODO: register ExtendedTextNode + replacement on nested editors
+    const nodeType = editor.hasNode(ExtendedTextNode) ? ExtendedTextNode : TextNode;
+
+    useKoenigTextEntity(
         getTKMatch,
         TKNode,
         createTKNode,
+        nodeType
     );
 
+    // we only want to render TK indicators for the top level editor
+    if (parentEditorNodeKey) {
+        return null;
+    }
+
     const editorRoot = editor.getRootElement();
     const editorRootParent = editor.getRootElement()?.parentElement;
 
     if (!editorRootParent) {
         return null;
     }
 
-    const tkParentNodesMap = {};
+    const TKIndicators = Object.entries(tkNodeMap).map(([parentKey, nodeKeys]) => {
+        const parentContainer = editor.getElementByKey(parentKey);
 
-    if (tkNodes.length > 0) {
-        editor.getEditorState().read(() => {
-            tkNodes.forEach((tkNode) => {
-                const parentKey = tkNode.getParent().getKey();
-
-                // prevent duplication, add node keys to existing indicator
-                // for nodes that are contained in the same parent
-                if (tkParentNodesMap[parentKey]) {
-                    tkParentNodesMap[parentKey].push(tkNode.getKey());
-                    return;
-                }
-
-                tkParentNodesMap[parentKey] = [tkNode.getKey()];
-            });
-        });
-    }
+        if (!parentContainer) {
+            return false;
+        }
 
-    const TKIndicators = Object.entries(tkParentNodesMap).map(([parentKey, nodeKeys]) => {
         return (
             <TKIndicator
                 key={parentKey}
-                containingElement={editor.getElementByKey(parentKey)}
                 editor={editor}
                 nodeKeys={nodeKeys}
+                parentKey={parentKey}
                 rootElement={editorRoot}
             />
         );
-    });
+    }).filter(Boolean);
```

Comment:
What is this appended for?

Reply:
The map over the tkNodeMap will return `false` if we can't get the parent container which will sometimes happen depending on when the re-render occurs. The `.filter(Boolean)` is there to strip out any empty/falsy entries from the resulting array so React doesn't have issues when we try to render the array as a component list.